aiida_quantumespresso/parsers/parse_raw/pw.py

```diff
@@ -743,6 +743,14 @@ def parse_stdout(stdout, input_parameters, parser_options=None, parsed_xml=None)
                     parsed_data['fermi_energy' + units_suffix] = default_energy_units
                 except Exception:
                     logs.warning.append('Error while parsing Fermi energy from the output file.')
+            elif 'Gaussian-smeared nuclei' in line:
+                try:
+                    value_potential_shift = float(line.split()[-2])
+                    unit_potential_shift = line.split()[-1]
+                    parsed_data['environ_potential_shift'] = value_potential_shift
+                    parsed_data['environ_unit_potential_shift'] = unit_potential_shift
+                except Exception:
+                    pass
```

Comment:
Add a log here as well?

Reply:
Thanks Giovanni! I have added it to the log in the most recent commit - thanks!

https://github.com/sudarshanv01/aiida-quantumespresso/blob/894b95a9e44fda4875b0a4f3fcd580bd40242e29/aiida_quantumespresso/parsers/parse_raw/pw.py#L753

Reply:
Yup, I guess it is specific to ENVIRON calculations, right now it says `Not an Environ calculation with an open boundary condition`, but this might show up for all calculations, which might be a bit annoying? Is there a better way to have a log message here without it popping up every time - and only for ENVIRON calculations?

Reply:
Sure, just add a conditional to the `elif 'Gaussian-smeared nuclei' in line` to only run when it is an ENVIRON calculation. How to determine whether we actually are parsing an ENVIRON question depends. Probably you could parse the stdout for some flag, but this is really fragile. If the XML doesn't contain some switch that we can rely on (which would be the best and most robust solution) you can use the input parameters. If I understand correctly, if you want to run an ENVIRON calculation, you have to add a particular key to the `parameters` input node. You can check for that at the beginning of the parser. Something like
```python
is_environ = 'ENVIRON' in self.inputs.parameters.get_dict()
```

and then the parsing line would become
```python
elif is_environ and 'Gaussian-smeared nuclei' in line:
```

Reply:
Sure, absolutely - just put this change in. In the current form I would need to pass an extra argument of settings to the `parse_raw` function. I could also just send `is_environ` as an argument, but maybe it is more complete to pass the `settings` dictionary, just like `parameters`?